### PR TITLE
[DEV-2801] inherit purpose when creating observation table

### PR DESCRIPTION
**Pull request by vikua**
## Description

inherit purpose when creating observation table

## Related Issue

[DEV-2801](https://featurebyte.atlassian.net/browse/DEV-2801)

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

[DEV-2801]: https://featurebyte.atlassian.net/browse/DEV-2801?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by YS-L — COMMENTED** [review]
Is it possible to write a test for this?

**Review by YS-L — APPROVED** [review]

**Comment by vikua** [comment]
@YS-L I don't really see such a possibility. I feel like tests for observation table part are in featurebyte-app repo 
That's why I created a PR https://github.com/featurebyte/featurebyte-app/pull/1006, where I found a test which fails and will be fixed once I bump featurebyte. 

I think we should avoid spreading one feature across repos (e.g. the task is in featurebyte, route which triggers the task in featurebyte-app and the test as well).

**Comment by vikua** [comment]
@YS-L Or maybe we have some other unit tests for this which I'm not aware of - let me know :)

File: featurebyte/worker/task/target_table.py [hunk 1]
```diff
@@ -102,6 +106,7 @@ async def execute(self, payload: TargetTableTaskPayload) -> Any:
                 context_id=payload.context_id,
                 request_input=payload.request_input,
                 primary_entity_ids=primary_entity_ids,
+                purpose=purpose,
                 **additional_metadata,
             )
             await self.observation_table_service.create_document(observation_table)
```
Comment (gee-senbong) [inline featurebyte/worker/task/target_table.py:112]: Could we write similar test like those in https://github.com/featurebyte/featurebyte/blob/main/tests/unit/worker/task/test_target_table.py?

**Comment by YS-L** [comment]
I see. It seems fine to me if there is test coverage in the app repo.

**Pull request merged**